test/runtests.jl

```diff
@@ -13,53 +13,46 @@ using LinearSolve: RFLUFactorization, LUFactorization, KrylovJL_GMRES
 using Aqua: Aqua
 
 """
-    experimental_order_of_convergence(prob, alg, dts, test_times;
-                                      only_first_index = false)
+    experimental_order_of_convergence(prob, alg, dts; test_time = nothing)
 
 Solve `prob` with `alg` and fixed time steps taken from `dts`, and compute
-the mean error at the times `test_times`.
-Return the associated experimental order of convergence.
+the errors at `test_time`. If`test_time` is not specified the error is computed 
+at the final time.
+Return the associated experimental orders of convergence.
 
 If `only_first_index == true`, only the first solution component is used
 to compute the error.
 """
-function experimental_order_of_convergence(prob, alg, dts, test_times;
+function experimental_order_of_convergence(prob, alg, dts; test_time = nothing,
                                            only_first_index = false)
     @assert length(dts) > 1
     errors = zeros(eltype(dts), length(dts))
     analytic = t -> prob.f.analytic(prob.u0, prob.p, t)
 
-    for (i, dt) in enumerate(dts)
-        sol = solve(prob, alg; dt = dt, adaptive = false)
+    if isnothing(test_time)
         if only_first_index
-            errors[i] = mean(test_times) do t
-                norm(sol(t; idxs = 1) - first(analytic(t)))
+            for (i, dt) in enumerate(dts)
+                sol = solve(prob, alg; dt = dt, adaptive = false, save_everystep = false)
+                errors[i] = norm(sol.u[end][1] - analytic(sol.t[end])[1])
             end
         else
-            errors[i] = mean(test_times) do t
-                norm(sol(t) - analytic(t))
```

Comment:
This changes the method to compute the error. Before, we used interpolation to several test times and computed the mean value of the errors. Now, only a single test time for interpolation is used. Why was this change necessary?

Reply:
First of all, computing the mean of orders did not work. For a third order method like `SSPMPRK43` you have to use rather large time steps to see the order, compared to the rather small time steps you need to see the order for a 1st order method like `MPE`. Thus,  to avoid to choose different vectors `dts`  for different schemes, I replaced computing the mean by looking for consecutive elements in the `orders`, which approximately show the expected order.

```julia
julia> orders = experimental_order_of_convergence(prob_pds_linmod, SSPMPRK43(), dts)
14-element Vector{Float64}:
 10.575835463004633
 -1.203554957240878
  2.1384358856941743
  2.8137438355112807
  3.128271451889756
  3.2210020871836496
  3.1941336864422776
  3.1465663723675665
  3.2872640260532053
  0.38495582792682975
 -1.048721232759049
 -1.0770825478215453
 -0.9936791864852714
 -0.8635662160256327

julia> mean(orders)
1.907400321124357

julia> orders = experimental_order_of_convergence(prob_pds_linmod, MPE(), dts)
14-element Vector{Float64}:
 2.586743267679417
 2.3989366692663556
 1.971835919757691
 1.5679744191512777
 1.3026043904022473
 1.1550539791231726
 1.0782866174655852
 1.0393061433904316
 1.0196897478419122
 1.0098536168109276
 1.0049371280440902
 1.002462681770976
 1.0012811111136874
 1.0000064071933645

julia> mean(orders)
1.3670694356436524
```
After this change, some tests were still failing, although they shouldn't. So, regarding the interpolation and mean error computation,  I thought it would be cleaner to go through the individual times and calculate the errors separately. I also hoped that this would reduce the number of failing tests. Now the tests in `runtests.jl` loop through the different times is `time_steps` and compute the errors and order for each time separately.